src/python/pants/engine/process.py

```diff
@@ -470,6 +472,30 @@ def __init__(
         super().__init__(msg)
 
 
+class BashBinary(BinaryPath):
+    """The `bash` binary."""
+
+    DEFAULT_SEARCH_PATH = SearchPath(("/usr/bin", "/bin", "/usr/local/bin"))
+
+
+@rule(desc="Finding the `bash` binary", level=LogLevel.DEBUG)
+async def find_bash(search_path: SearchPath) -> BashBinary:
+    request = BinaryPathRequest(
+        binary_name="bash", search_path=search_path, test=BinaryPathTest(args=["--version"])
+    )
+    paths = await Get(BinaryPaths, BinaryPathRequest, request)
+    first_path = paths.first_path
+    if not first_path:
+        raise BinaryNotFoundError(request, rationale="execute bash scripts")
```

Comment:
I recommend the old wording. With this, a user may think "Okay, but I'm not trying to execute any bash scripts?" It's an implementation detail that Pants is using bash scripts to run things.

Reply:
I mean - we already do this in spades. No user wants a PEX for anything but package but we print "*.pex" everywhere.

That aside, we're trading confusion. If my rule requests a BashBinary and when it can't be found it hears "use it to locate other executables" ... that rule will scratch it's head.

I'll parametrize the rationale.